test/correctness/simd_op_check.cpp

```diff
@@ -1,13 +1,12 @@
 #include "Halide.h"
+#include "simd_op_check.h"
```

Comment:
I presume this refactoring was done so you could have a new simd_op_check_hvx.cpp test, but it's not clear to me why it's a separate test now -- can you add some comment to this PR about why that's desirable?

Reply:
The Halide buildbots run correctness tests by setting ```HL_TARGET``` to the same value as ```HL_JIT_TARGET```. When testing hexagon, they are host-hvx_128. This works for all correctness tests except simd_op_check.cpp which is not an 'offload' tests at all - hexagon tests are run if the host processor is hexagon. In other words, it'll run hexagon tests only if ```HL_TARGET``` has a hexagon host arch like ```hexagon-32-noos-hvx_128```.

As a result the buildbot never really tested HVX correctly and over time we have discovered a number of failures in simd_op_check that the buildbots didn't catch simply because they weren't testing simd_op_check with the right target for HVX.

One solution would be to change the buildbot recipe but after I looked at the buildbot recipe, I rejected that idea because it would mean one exception for one test on architecture. Instead, I decided to refactor this way, so HVX is free to do its own thing and the buildbot can keep its model of ```HL_TARGET=HL_JIT_TARGET```

Reply:
Could you add this detail as a comment in simd_op_check_hvx.cpp, so that future readers won't be tempted to recombine them without realizing this?

Reply:
Done.